### PR TITLE
refactor more GitRepository code to simplify parameters passed in

### PR DESCRIPTION
**Pull request by jonmoter**
/cc @zendesk/runway 

### Risks
 - Confirm that logic still works for running git commands

### CODE REVIEW

**Comment by grosser** [comment]
:+1:

File: app/models/git_repository.rb [hunk 1]
```diff
@@ -1,31 +1,39 @@
...
+  attr_accessor :executor
 
   # The directory in which repositories should be cached.
   cattr_accessor(:cached_repos_dir, instance_writer: false) do
     Rails.application.config.samson.cached_repos_dir
   end
 
-  def initialize(repository_url:, repository_dir:)
+  def initialize(repository_url: nil, repository_dir: nil, executor: nil)
+    raise ArgumentError.new("GitRepository.new: repository_url is required") if repository_url.blank?
```
Comment (sandlerr) [inline app/models/git_repository.rb]: these were already required. Just testing for empty string?

Comment (jonmoter) [inline app/models/git_repository.rb]: Ah, I didn't realize the `repository_url:,` syntax meant that it was required.  Will revert.

**Comment by sandlerr** [comment]
:+1:

**Pull request merged**